src/prefect/artifacts.py

```diff
@@ -0,0 +1,160 @@
+from typing import Union
+
+from prefect import context, Client
+
+
+def running_with_backend() -> bool:
+    """
+    Determine if running in context of a backend. This is always true when running
+    using the `CloudTaskRunner`.
+
+    Returns:
+        - bool: if `running_with_backend` is set in context
+    """
+    return bool(context.get("running_with_backend"))
+
+
+def create_link(link: str) -> Union[str, None]:
+    """
+    Create a link artifact
+
+    Note: The functionality here is experimental, and may change between
+    versions without notice. Use at your own risk.
+
+    Args:
+        - link (str): the link to post
+
+    Returns:
+        - str: the task run artifact ID
+    """
+    if not running_with_backend():
+        return None
+
+    client = Client()
+    return client.create_task_run_artifact(
+        task_run_id=context.get("task_run_id"), kind="link", data={"link": link}
+    )
+
+
+def update_link(task_run_artifact_id: str, link: str) -> bool:
+    """
+    Update an existing link artifact. This function will replace the current link
+    artifact with the new link provided.
+
+    Note: The functionality here is experimental, and may change between
+    versions without notice. Use at your own risk.
+
+    Args:
+        - task_run_artifact_id (str): the ID of an existing task run artifact
+        - link (str): the new link to update the artifact with
+
+    Returns:
+        - bool: whether or not the request was successful
+    """
+    if not running_with_backend():
+        return False
```

Comment:
Could be nice to have as a decorator? Should there be a UserWarning involved?

Reply:
Yeah can make a decorator, was getting some weird documentation errors so switched to this but on biggie

Reply:
I think it mostly makes sense if it's more than one line of repeated handling

Reply:
Discussed w/ @jcrist, skipping decorator for now